lib/kitchen/directions/bake_figure.rb

```diff
@@ -5,29 +5,48 @@ module Directions
     module BakeFigure
       def self.v1(figure:, number:, cases: false)
         warn 'warning! exclude unnumbered figures from `BakeFigure` loop' if figure.unnumbered?
-        figure.wrap(%(<div class="os-figure#{' has-splash' if figure.has_class?('splash')}">))
+        figure.wrap(%(<div class="os-figure#{if figure.has_class?('splash')
+                                               ' has-splash'
+                                             elsif figure.has_class?('mechanism-figure')
+                                               ' has-mechanism-figure'
+                                             end}">))
 
         # Store label information
         figure.target_label(label_text: 'figure', custom_content: number, cases: cases)
-
         title = figure.title&.cut
         caption = figure.caption&.cut
         caption&.name = 'span'
         caption&.add_class('os-caption')
         caption&.remove_attribute('id') # only needed for pl books, where captions have id
 
-        figure.append(sibling:
-          <<~HTML.chomp
-            <div class="os-caption-container">
-              <span class="os-title-label">#{I18n.t("figure#{'.nominative' if cases}")} </span>
-              <span class="os-number">#{number}</span>
-              #{"<span class=\'os-divider\'> </span>" if title}
-              #{"<span class=\'os-title\' data-type=\'title\' id=\"#{title.id}\">#{title.children}</span>" if title}
-              #{"<span class=\'os-divider\'> </span>" if caption}
-              #{caption}
-            </div>
-          HTML
-        )
+        if figure.has_class?('mechanism-figure')
+          figure.prepend(sibling:
+            <<~HTML
+              <div class="os-caption-title-container">
+                <span class="os-title-label">#{I18n.t("figure#{'.nominative' if cases}")} </span>
+                <span class="os-number">#{number}</span>
+                #{"<span class=\'os-divider\'> </span>" if title}
+                #{"<span class=\'os-title\' data-type=\'title\' id=\"#{title.id}\">#{title.children}</span>" if title}
+              </div>
+              <div class="os-caption-text-container">
+                #{caption}
+              </div>
+            HTML
+          )
+        else
+          figure.append(sibling:
+            <<~HTML.chomp
+              <div class="os-caption-container">
+                <span class="os-title-label">#{I18n.t("figure#{'.nominative' if cases}")} </span>
+                <span class="os-number">#{number}</span>
+                #{"<span class=\'os-divider\'> </span>" if title}
+                #{"<span class=\'os-title\' data-type=\'title\' id=\"#{title.id}\">#{title.children}</span>" if title}
+                #{"<span class=\'os-divider\'> </span>" if caption}
+                #{caption}
+              </div>
+            HTML
```

Comment:
is the only difference here prepending or appending? I would suggest creating a `caption_html` variable and doing an

```
if figure.has_class?('mechanism-figure')
  figure.prepend(sibling: caption_html)
else
  figure.append(sibling: caption_html)
end
```

Reply:
oh, i see now. oh well. it might be possible to reuse a few lines -- these lines seem to be identical in both

```
<span class="os-title-label">#{I18n.t("figure#{'.nominative' if cases}")} </span>
<span class="os-number">#{number}</span>
#{"<span class=\'os-divider\'> </span>" if title}
#{"<span class=\'os-title\' data-type=\'title\' id=\"#{title.id}\">#{title.children}</span>" if title}
```

but i leave it up to you whether you think that's worth it